### ISSUE TITLE
Update to curl 8.11.1: Fixes enable to build curl with gcc-14 on macOS

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/ESMA-Baselibs#247]
TL;DR It looks like with XCode 16.2 and Brew `gcc-14`, we now *need* curl 8.11.1 to build, along with the `SDKROOT` fix via @scivision

---

### No changes

I recently tried building Baselibs with `gcc-14` from Brew. In doing so, it failed in the curl 8.10.1 build with many, many of:
```
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/sys/_types/_fd_def.h:32,
                 from /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/sys/types.h:187,
                 from curl_setup.h:64,
                 from altsvc.c:28:
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/unistd.h:744:1: error: 'deprecated' undeclared here (not in a function)
  744 | __OS_AVAILABILITY_MSG(ios,deprecated=10.0,"syscall(2) is unsupported; "
      | ^~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/unistd.h:744:1: error: 'message' undeclared here (not in a function)
  744 | __OS_AVAILABILITY_MSG(ios,deprecated=10.0,"syscall(2) is unsupported; "
      | ^~~~~~~~~~~~~~~~~~~~~
```

Now, I haven't tried in a while, but this seems to have happened after I updated to XCode 16.2? Or maybe an update to Brew? Dunno.

My first suspicion was the @scivision `SDKROOT` issue (see https://gist.github.com/scivision/d69faebbc56da9714798087b56de925a), but I already had:
```lua
-- Bug fix for XCode 16
setenv("SDKROOT","/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk")
```
in my modulefile. This is seen by the fact that the above error messages have MacOSX14 in them. So it was being passed down. And, well, I tried the reproducer in the gist above and it works fine too.

### MacOSX15 SDKROOT
 
I did try building curl with other `SDKROOT` just to see, but no help. If I specifically use the 15 `SDKROOT`:
```
export SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk
```
then I get:
```
checking for suffix of executables...
checking whether we are cross compiling... configure: error: in '/Users/mathomp4/Baselibs/ESMA-Baselibs-8.7.0/src/curl':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use '--host'.
See 'config.log' for more details
make[1]: *** [curl.config] Error 77
make: *** [install] Error 2
```

and in `curl.config`:
```
In file included from conftest.c:9:
/Users/mathomp4/.homebrew/brew/Cellar/gcc/14.2.0_1/lib/gcc/current/gcc/aarch64-apple-darwin23/14/include-fixed/stdio.h:83:8: error: unknown type name 'FILE'
   83 | extern FILE *__stdinp;
      |        ^~~~
/Users/mathomp4/.homebrew/brew/Cellar/gcc/14.2.0_1/lib/gcc/current/gcc/aarch64-apple-darwin23/14/include-fixed/stdio.h:81:1: note: 'FILE' is defined in header '<stdio.h>'; this is probably fixable by adding '#include <stdio.h>'
   80 | #include <sys/_types/_seek_set.h>
  +++ |+#include <stdio.h>
   81 |
/Users/mathomp4/.homebrew/brew/Cellar/gcc/14.2.0_1/lib/gcc/current/gcc/aarch64-apple-darwin23/14/include-fixed/stdio.h:84:8: error: unknown type name 'FILE'
   84 | extern FILE *__stdoutp;
      |        ^~~~
/Users/mathomp4/.homebrew/brew/Cellar/gcc/14.2.0_1/lib/gcc/current/gcc/aarch64-apple-darwin23/14/include-fixed/stdio.h:84:8: note: 'FILE' is defined in header '<stdio.h>'; this is probably fixable by adding '#include <stdio.h>'
/Users/mathomp4/.homebrew/brew/Cellar/gcc/14.2.0_1/lib/gcc/current/gcc/aarch64-apple-darwin23/14/include-fixed/stdio.h:85:8: error: unknown type name 'FILE'
   85 | extern FILE *__stderrp;
      |        ^~~~
```
but that looks like the error reported by @scivision

### Update curl

Finally, just in case @bagder fixed something, I grabbed the latest curl, 8.11.1, and... it works! Looking around the changelog and I see https://github.com/curl/curl/pull/15508 via @vszakats so huzzah!

And, yep, we still need the `SDKROOT` fix it seems. Trying `MacOSX15` broke the same as before.

**Comment by mathomp4** [comment]
Closed via #248 and #249
